docs/Golden-Gate/Bridge-Walkthrough.rst

```diff
@@ -0,0 +1,144 @@
+top-level of our chip.
+
+We first define a class that captures the target-side interface of the Bridge:
+
+.. literalinclude:: ../../sim/firesim-lib/src/main/scala/bridges/UARTBridge.scala
+    :language: scala
+    :start-after: DOC include start: UART Bridge Target-Side Interface
+    :end-before: DOC include end: UART Bridge Target-Side Interface
+
+Here, we define a case class that carries additional metadata to the host-side
+BridgeModule.  For UART, this is simply the clock-division required to produce the
+baudrate:
+
+.. literalinclude:: ../../sim/firesim-lib/src/main/scala/bridges/UARTBridge.scala
+    :language: scala
+    :start-after: DOC include start: UART Bridge Constructor Arg
+    :end-before: DOC include end: UART Bridge Constructor Arg
+
+Finally, we define the actual target-side module (specifically, a BlackBox):
+
+.. literalinclude:: ../../sim/firesim-lib/src/main/scala/bridges/UARTBridge.scala
+    :language: scala
+    :start-after: DOC include start: UART Bridge Target-Side Module
+    :end-before: DOC include end: UART Bridge Target-Side Module
+
+To make it easier to instantiate our target-side module, we've also defined an
+optional companion object:
+
+.. literalinclude:: ../../sim/firesim-lib/src/main/scala/bridges/UARTBridge.scala
+    :language: scala
+    :start-after: DOC include start: UART Bridge Companion Object
+    :end-before: DOC include end: UART Bridge Companion Object
+
+That completes the target-side definition.
+
+UART Host-Side BridgeModule
+++++++++++++++++++++++++++++
+
+The remainder of the file is dedicated to the host-side BridgeModule
+definition. Here we have to process tokens generated by the target, and
+expose a memory-mapped interface to the bridge driver.
+
+Inspecting the top of the class:
+
+.. literalinclude:: ../../sim/firesim-lib/src/main/scala/bridges/UARTBridge.scala
+    :language: scala
+    :start-after: DOC include start: UART Bridge Header
+    :end-before: DOC include end: UART Bridge Header
+
+
+Most of what follows is responsible for modeling the timing of the UART.
+As a bridge designer, you're free to take as many host-cycles as you need to
+process tokens. In simpler models, like this one, it's often easiest to write
+logic that operates in a single cycle but gate state-updates using a
+"fire" signal that is asserted when the required tokens are available.
+
+Now, we'll skip to the end to see how to add registers to the simulator's memory map
+that can be accessed using MMIO from bridge driver.
+
+.. literalinclude:: ../../sim/firesim-lib/src/main/scala/bridges/UARTBridge.scala
+    :language: scala
+    :start-after: DOC include start: UART Bridge Footer
+    :end-before: DOC include end: UART Bridge Footer
+
+
+UART Bridge Driver
+++++++++++++++++++
+
+To complete our host-side definition, we need to define a CPU-hosted bridge driver.
+Bridge Drivers extend the ``bridge_driver_t`` interface, which declares 5 virtual methods
```

Comment:
Maybe mention the role of each virtual method, especially since UART doesn't define init and finish?

Reply:
Good idea. I can just inline the header for the interface.

Reply:
Does the header include comments of what each of them should do?